Add podcast season 3, episode 3 (#5582)
sourcegraph/about: content/podcast/3.3.md

---
title: 'John Kodumal, CTO and Co-founder of LaunchDarkly'
publishDate: 2022-07-14T10:00-10:00
tags: [podcast]
authors:
  - name: John Kodumal
  - name: Beyang Liu
slug: john-kodumal
published: false
description: "Beyang sits down with John Kodumal, CTO and co-founder of LaunchDarkly. LaunchDarkly is a SaaS feature management platform for developers. Beyang and John discuss John’s early years in programming, his time at Covarity and Atlassian, and the ins and outs of founding and managing LaunchDarkly."
audioSrc: "https://www.buzzsprout.com/1097978/10962273-john-kodumal-cto-and-co-founder-of-launchdarkly.mp3"
videoID: "9d5HALmEMNs"
---

Beyang sits down with John Kodumal, CTO and co-founder of [LaunchDarkly](https://launchdarkly.com/). LaunchDarkly is a SaaS feature management platform for developers that allows them to iterate and get code into production quickly and safely by separating feature rollout and code deployment.

> John begins by talking about his first experiences with computers and programing in the 80s, including teaching himself to us a Dvorak keyboard in the first grade, experimenting with BBS in elementary school, and programming his TI-92 in BASIC to make a shell program so that he could use Reverse Polish Notation (RPN) on it in high school. John shares how he pursued his interest in programming languages throughout higher education and then discusses his employment experiences at Coverity and Atlassian. He talks about how the lessons and experiences from his prior jobs ultimately led him to found LaunchDarkly in 2014 with former classmate, Edith Harbaugh.

Reviewer: ```suggestion
John begins by talking about his first experiences with computers and programming in the 80s, including teaching himself to use a Dvorak keyboard in the first grade, experimenting with Bulletin Board Systems in elementary school, and programming his TI-92 in BASIC to let him use Reverse Polish Notation (RPN). John shares how he pursued his interest in programming languages throughout higher education and shares stories and lessons from his time at Coverity and Atlassian. He talks about how the lessons and experiences from his prior jobs ultimately led him to found LaunchDarkly in 2014 with his former college lab partner, Edith Harbaugh.
```

> John dives into how did he first got into feature toggles and feature flags, and then talks about the engineering challenges LaunchDarkly has encountered. John concludes by sharing how he has witnessed LaunchDarkly impact the developer experience and the ongoing, transformational benefits of utilizing their feature management platform.

Reviewer: ```suggestion
John dives into how he and Edith settled on the problem area of feature toggles and feature flags, and then talks about the engineering challenges LaunchDarkly has encountered. John concludes by sharing how he has witnessed LaunchDarkly impact the developer experience and the ongoing, transformational benefits of utilizing their feature management platform for their customers.
```

<center>
<iframe width="560" height="315" src="https://www.youtube.com/embed/9d5HALmEMNs" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</center>

<center className="pt-2 pb-4">
    <strong>Subscribe:</strong>
    <a href="https://www.youtube.com/playlist?list=PL6zLuuRVa1_jf5GDl61SvEOXvwvKS1IXA" style={{paddingLeft: '0.2em'}}>YouTube</a>
    <a href="https://podcasts.apple.com/us/podcast/the-sourcegraph-podcast/id1516219009" style={{paddingLeft: '0.2em'}}>Apple</a>
    <a href="https://open.spotify.com/show/1YlDYvCxNB7jAndbZPt5a6" style={{paddingLeft: '0.2em'}}>Spotify</a>
    <a href="https://podcasts.google.com/?feed=aHR0cHM6Ly9mZWVkcy5idXp6c3Byb3V0LmNvbS8xMDk3OTc4LnJzcw==" style={{paddingLeft: '0.2em'}}>Google</a>
    <a href="https://feeds.buzzsprout.com/1097978.rss" style={{paddingLeft: '0.2em'}}>RSS</a>
</center>

<hr className="py-4" />

## Show Notes

>  * John tells the story of his introduction to computers, beginning with BBS and CompuServe in elementary school, and his first forays into programming with BASIC and Turbo Pascal in the early 80s (00:24-7:55)

Reviewer: ```suggestion
 * John tells the story of his introduction to computers, beginning with BBSes and CompuServe in elementary school, and his first forays into programming with BASIC and Turbo Pascal in the early 80s (00:24-7:55)
```

* John talks about his interest in programming languages that fueled his pursuits in higher education (7:55-13:32)

* Beyang and John discuss the current wave of low code tools and the direction it’s moving (13:32-16:40)

* John talks about his work experience prior to starting LaunchDarkly, including a stint at Coverity, and then at Atlassian which he credits as being where he learned to build a company and scale teams (16:40-20:54) 

* Beyang and John talk about the future of static analysis (20:54-24:57)

> * John delves into his time at Atlassian and explains what it was like witnessing the company’s growth from a 150 person operation to IPO, and emphasizes Atlassian’s high caliber culture of collaboration that continues to influence his management of LaunchDarkly (24:57-27:32)

Reviewer: ```suggestion
* John delves into his time at Atlassian and explains what it was like witnessing the company’s growth from a 150 person operation to IPO, and emphasizes Atlassian’s high caliber culture of collaboration that continues to influence his leadership style at LaunchDarkly (24:57-27:32)
```

> * John talks about some of Atlassian’s potential competitors through the years including Fog Creek and GitHub, and Beyang and John imagine what could be in store for workflow/change management (27:32-35:57)

Reviewer: ```suggestion
* John talks about some of Atlassian’s positioning relative to other developer-focused companies like Fog Creek and GitHub, and Beyang and John imagine what could be in store for workflow/change management (27:32-35:57)
```

* John shares what inspired him to leave Atlassian and found LaunchDarkly with Edith Harbaugh, then talks about the challenges of starting a company in the developer tools space in the early 2010s (35:57-41:02)

* In answering the question “How did you get into feature toggles and feature flags” John talks about the internal flagging tool he helped to build at Atlassian, and how his experience with the pain points on that internal platform influenced his idea to start LaunchDarkly, and then relates a story of hacking Optimizely to temporarily patch a bug in the Atlassian purchasing system (41:02-44:21)

* John talks about LaunchDarkly’s first customers and the realization that most companies were already feature flagging, just doing so badly, and Beyang and John discuss targeting enterprises early in the life of a dev tools start-up versus mainly SMB (44:21-49:04)

> * John discusses the engineering challenges that LaunchDarkly encountered in terms of scale and reliability including: eliminating remote calls, getting updates pushed over via persistent HTTPS connections, and optimizing individual user context (49:04-56:49)

Reviewer: ```suggestion
* John discusses the engineering challenges that LaunchDarkly has encountered in terms of scale and reliability, and solutions such as eliminating remote calls, getting updates pushed over persistent HTTPS connections, and optimizing individual user context (49:04-56:49)
```
